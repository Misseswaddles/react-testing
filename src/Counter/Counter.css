.Counter {
    font-size: 2em;
}

.Counter__button {
    border: none;
    font-size: 2em;
    padding: 10px 30px;
    margin: 10px;
    cursor: pointer;
}

.Counter__button--increment {
    background-color: green;
}

.Counter__button--decrement {
    background-color: red;
}