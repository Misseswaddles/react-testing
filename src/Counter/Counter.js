import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Counter.css';

class Counter extends Component {
    constructor(props) {
        super(props);
        this.state = {
            value: props.initialValue
        };
    }

    increment = () => {
        this.setState((prevState) => {
            const newValue = prevState.value + 1;
            return { value: newValue };
        });
    }

    decrement = () => {
        this.setState((prevState) => {
            const newValue = prevState.value - 1;
            return { value: newValue };
        });
    }

    render() {
        return (
            <div className="Counter">
                <h2>{this.state.value}</h2>
                <button className="Counter__button Counter__button--decrement" onClick={this.decrement}>-</button>
                <button className="Counter__button Counter__button--increment" onClick={this.increment}>+</button>
            </div>
        );
    }
}

Counter.propTypes = {
    initialValue: PropTypes.number
};

Counter.defaultProps = {
    initialValue: 0
};

export default Counter;