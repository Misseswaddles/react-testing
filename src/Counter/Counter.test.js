import React from 'react';
import { shallow, mount } from 'enzyme';
import Counter from './Counter';

describe('Counter', () => {
    it('Should increment when clicking on the increment button', () => {
        const wrapper = mount(<Counter />);

        // console.log(wrapper.find('button'));

        expect(wrapper.find('button')).toHaveLength(2);
        wrapper.find('.Counter__button--increment').simulate('click').simulate('click');
        expect(wrapper).toMatchSnapshot();
    });

    it('Should decrement when clicking on the decrement button', () => {
        const wrapper = mount(<Counter />);

        // console.log(wrapper.find('button'));

        expect(wrapper.find('button')).toHaveLength(2);
        wrapper.find('.Counter__button--decrement').simulate('click').simulate('click');
        expect(wrapper).toMatchSnapshot();
    });
});