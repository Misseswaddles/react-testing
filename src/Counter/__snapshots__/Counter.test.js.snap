// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Should increment when clicking on the increment button 1`] = `
<Counter
  initialValue={0}
>
  <div
    className="Counter"
  >
    <h2>
      2
    </h2>
    <button
      className="Counter__button Counter__button--decrement"
      onClick={[Function]}
    >
      -
    </button>
    <button
      className="Counter__button Counter__button--increment"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</Counter>
`;
